src/hooks/api/index.ts

```diff
@@ -63,16 +63,17 @@ export function useAccountsMonthlyTotals(): SWRResponse<queries.MonthlyTotals> {
     fetcher(PriceDB.getTodayQuotes, '/api/prices/today'),
   );
 
-  const key = (accounts && todayPrices) ? [
-    '/api/accounts/monthly-totals',
-    // We need to recompute when accounts and prices change
-    Object.keys(accounts),
-    Object.keys(todayPrices.keys),
-  ] : null;
+  // This makes sure that every time new accounts appear we recalculate
+  // the monthly totals.
+  //
+  // Not only this but re-using the same key has super poor performance
+  // for some reason
+  const key = (accounts && todayPrices)
+    ? `/api/monthly-totals?ids=${Object.keys(accounts).join(',')}` : null;
```

Comment:
not a big difference from before but prefer to have consistent type for keys (strings).